Error on the newest SF version

```
[13:17:13] [Server thread/ERROR]: Error occurred while enabling SoundMuffler vDEV - 12 (git 7303b3a9) (Is it up to date?)
java.lang.NoSuchMethodError: 'void me.mrCookieSlime.Slimefun.api.Slimefun.registerResearch(me.mrCookieSlime.Slimefun.Objects.Research, org.bukkit.inventory.ItemStack[])'
	at dev.j3fftw.soundmuffler.SoundMuffler.onEnable(SoundMuffler.java:30) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.16.1.jar:git-Paper-101]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[patched_1.16.1.jar:git-Paper-101]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[patched_1.16.1.jar:git-Paper-101]
	at org.bukkit.craftbukkit.v1_16_R1.CraftServer.enablePlugin(CraftServer.java:500) ~[patched_1.16.1.jar:git-Paper-101]
	at org.bukkit.craftbukkit.v1_16_R1.CraftServer.enablePlugins(CraftServer.java:414) ~[patched_1.16.1.jar:git-Paper-101]
	at net.minecraft.server.v1_16_R1.MinecraftServer.loadWorld(MinecraftServer.java:471) ~[patched_1.16.1.jar:git-Paper-101]
	at net.minecraft.server.v1_16_R1.DedicatedServer.init(DedicatedServer.java:241) ~[patched_1.16.1.jar:git-Paper-101]
	at net.minecraft.server.v1_16_R1.MinecraftServer.v(MinecraftServer.java:940) ~[patched_1.16.1.jar:git-Paper-101]
	at net.minecraft.server.v1_16_R1.MinecraftServer.lambda$a$0(MinecraftServer.java:177) ~[patched_1.16.1.jar:git-Paper-101]
	at java.lang.Thread.run(Thread.java:832) [?:?]
```

Plugin doesn't work
Minecraft Version: 1.16.1
CS-CoreLib Version: 88
Slimefun Version: 588
Plugin Version: 12

@J3fftw1

I think I just reproduced this issue on my server. It's happening since a recent SlimeFun update.
Apparently something was changed in the SlimeFun code and this addon has not yet been updated to work with those changes.

```
[15:06:22 ERROR]: Error occurred while enabling SoundMuffler vDEV - 12 (git 7303b3a9) (Is it up to date?)
java.lang.NoClassDefFoundError: me/mrCookieSlime/Slimefun/Objects/Research
        at dev.j3fftw.soundmuffler.SoundMuffler.onEnable(SoundMuffler.java:30) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.16.1.jar:git-Paper-113]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[patched_1.16.1.jar:git-Paper-113]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[patched_1.16.1.jar:git-Paper-113]
        at org.bukkit.craftbukkit.v1_16_R1.CraftServer.enablePlugin(CraftServer.java:500) ~[patched_1.16.1.jar:git-Paper-113]
        at org.bukkit.craftbukkit.v1_16_R1.CraftServer.enablePlugins(CraftServer.java:414) ~[patched_1.16.1.jar:git-Paper-113]
        at net.minecraft.server.v1_16_R1.MinecraftServer.loadWorld(MinecraftServer.java:471) ~[patched_1.16.1.jar:git-Paper-113]
        at net.minecraft.server.v1_16_R1.DedicatedServer.init(DedicatedServer.java:241) ~[patched_1.16.1.jar:git-Paper-113]
        at net.minecraft.server.v1_16_R1.MinecraftServer.v(MinecraftServer.java:940) ~[patched_1.16.1.jar:git-Paper-113]
        at net.minecraft.server.v1_16_R1.MinecraftServer.lambda$a$0(MinecraftServer.java:177) ~[patched_1.16.1.jar:git-Paper-113]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_241]
Caused by: java.lang.ClassNotFoundException: me.mrCookieSlime.Slimefun.Objects.Research
        at java.net.URLClassLoader.findClass(Unknown Source) ~[?:1.8.0_241]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:171) ~[patched_1.16.1.jar:git-Paper-113]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:100) ~[patched_1.16.1.jar:git-Paper-113]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_241]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_241]
        ... 11 more
```

--- a/pom.xml
+++ b/pom.xml
@@ -33,7 +33,7 @@
         <dependency>
             <groupId>com.github.TheBusyBiscuit</groupId>
             <artifactId>Slimefun4</artifactId>
-            <version>73401bae08</version>
+            <version>76362b2acb</version>
             <scope>provided</scope>
             <exclusions>
                 <exclusion>

--- a/pom.xml
+++ b/pom.xml
@@ -33,7 +33,7 @@
         <dependency>
             <groupId>com.github.TheBusyBiscuit</groupId>
             <artifactId>Slimefun4</artifactId>
-            <version>76362b2acb</version>
+            <version>RC-15</version>
             <scope>provided</scope>
             <exclusions>
                 <exclusion>

--- a/src/main/java/dev/j3fftw/soundmuffler/SoundMuffler.java
+++ b/src/main/java/dev/j3fftw/soundmuffler/SoundMuffler.java
@@ -26,10 +26,10 @@ public class SoundMuffler extends JavaPlugin implements SlimefunAddon {
         SoundMufflerMachine soundMufflerMachine = new SoundMufflerMachine();
         soundMufflerMachine.register(this);
 
-        Research research = new Research(new NamespacedKey(this, "sound_muffler"), 6912, "Sound Muffler", 11);
-        
-        research.addItems(soundMufflerMachine.getItem());
-        research.register();
+        new Research(new NamespacedKey(this, "sound_muffler"),
+                     6912, "Sound Muffler", 11)
+        .addItems(soundMufflerMachine.getItem())
+        .register();
     }
 
 

--- a/src/main/java/dev/j3fftw/soundmuffler/SoundMuffler.java
+++ b/src/main/java/dev/j3fftw/soundmuffler/SoundMuffler.java
@@ -1,9 +1,8 @@
 package dev.j3fftw.soundmuffler;
 
 import io.github.thebusybiscuit.slimefun4.api.SlimefunAddon;
+import io.github.thebusybiscuit.slimefun4.core.researching.Research;
 import me.mrCookieSlime.Slimefun.Objects.Category;
-import me.mrCookieSlime.Slimefun.Objects.Research;
-import me.mrCookieSlime.Slimefun.api.Slimefun;
 import me.mrCookieSlime.Slimefun.bstats.bukkit.Metrics;
 import me.mrCookieSlime.Slimefun.cscorelib2.item.CustomItem;
 import org.bukkit.Material;
@@ -27,8 +26,9 @@ public class SoundMuffler extends JavaPlugin implements SlimefunAddon {
         SoundMufflerMachine soundMufflerMachine = new SoundMufflerMachine();
         soundMufflerMachine.register(this);
 
-        Slimefun.registerResearch(new Research(new NamespacedKey(this, "sound_muffler"), 6912, "Sound Muffler", 11),
-            soundMufflerMachine.getItem());
+        Research research = new Research(new NamespacedKey(this, "sound_muffler"), 6912, "Sound Muffler", 11);
+        research.addItems(soundMufflerMachine.getItem());
+        research.register();
     }
 
 

--- a/src/main/java/dev/j3fftw/soundmuffler/SoundMuffler.java
+++ b/src/main/java/dev/j3fftw/soundmuffler/SoundMuffler.java
@@ -27,6 +27,7 @@ public class SoundMuffler extends JavaPlugin implements SlimefunAddon {
         soundMufflerMachine.register(this);
 
         Research research = new Research(new NamespacedKey(this, "sound_muffler"), 6912, "Sound Muffler", 11);
+        
         research.addItems(soundMufflerMachine.getItem());
         research.register();
     }

--- a/src/main/java/dev/j3fftw/soundmuffler/SoundMufflerMachine.java
+++ b/src/main/java/dev/j3fftw/soundmuffler/SoundMufflerMachine.java
@@ -2,6 +2,7 @@ package dev.j3fftw.soundmuffler;
 
 import io.github.thebusybiscuit.slimefun4.core.attributes.EnergyNetComponent;
 import io.github.thebusybiscuit.slimefun4.core.networks.energy.EnergyNetComponentType;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
 import me.mrCookieSlime.CSCoreLibPlugin.Configuration.Config;
 import me.mrCookieSlime.Slimefun.Lists.RecipeType;
 import me.mrCookieSlime.Slimefun.Lists.SlimefunItems;
@@ -9,7 +10,6 @@ import me.mrCookieSlime.Slimefun.Objects.SlimefunBlockHandler;
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.UnregisterReason;
 import me.mrCookieSlime.Slimefun.Objects.handlers.BlockTicker;
-import me.mrCookieSlime.Slimefun.SlimefunPlugin;
 import me.mrCookieSlime.Slimefun.api.BlockStorage;
 import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
 import me.mrCookieSlime.Slimefun.api.energy.ChargableBlock;

